controllers/helmrepository_controller.go

```diff
@@ -422,6 +427,13 @@ func (r *HelmRepositoryReconciler) reconcileSource(ctx context.Context, obj *sou
 	}
 	*chartRepo = *newChartRepo
 
+	// enable cache if applicable
+	if r.Cache != nil && chartRepo.IndexCache == nil {
+		chartRepo.SetMemCache(chartRepo.CachePath, r.Cache, r.TTL, func(event string) {
```

Comment:
we don't want the key `chartRepo.CachePath` to possibly be `""`.  First reason being that key collision. To avoid this, I think we should wait the last moment just after the `chartRepo.CachePath` has been generated to set the memory cache.

Reply:
If I understand it correctly currently this is immediately after [newChartRepo.CacheIndex() call](https://github.com/fluxcd/source-controller/pull/698/files#diff-c305fa3b711de20ff8e12a742562544c33b8d6eafb0145842324d89cdabf5fe5R418 ), which I believe we will get a non empty chartRepo.CachePath if that call succeed since it's from [this call](https://github.com/fluxcd/source-controller/blob/main/internal/helm/repository/chart_repository.go#L320).

Although, we can definitely add a double insurance call in [CacheIndex()](https://github.com/fluxcd/source-controller/blob/main/internal/helm/repository/chart_repository.go#L338) to make sure the f.Name() is non empty. What do you think?

Reply:
the code is correct you are right, but this still is not what we want.

Here: https://github.com/fluxcd/source-controller/blob/fe31ff9e773be73f4007ba689def9d3cda4a9dad/controllers/helmrepository_controller.go#L516 we will copy from memory to local disk, and set the `artifact` path.

That `artifact` path is the key used later on by the chart controller to construct a chartRepository here:
https://github.com/fluxcd/source-controller/blob/fe31ff9e773be73f4007ba689def9d3cda4a9dad/controllers/helmchart_controller.go#L564

and later load the index here:
https://github.com/fluxcd/source-controller/blob/fe31ff9e773be73f4007ba689def9d3cda4a9dad/internal/helm/repository/chart_repository.go#L371

We want the HelmRepository reconciler to use the same key (artifact path) so we can have a cache hit downstream

Reply:
done